pkg/cmd/image/save.go

```diff
@@ -0,0 +1,90 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/images/archive"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/idutil/imagewalker"
+	"github.com/containerd/nerdctl/pkg/platformutil"
+	"github.com/containerd/nerdctl/pkg/strutil"
+	"github.com/mattn/go-isatty"
+)
+
+// Save will save one or more images to a tar archive (streamed to STDOUT by default).
+func Save(ctx context.Context, client *containerd.Client, images []string, options types.ImageSaveOptions) error {
+	if options.Output != "" {
+		f, err := os.OpenFile(options.Output, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		options.Stdout = f
+	} else {
+		if isatty.IsTerminal(os.Stdout.Fd()) {
+			return fmt.Errorf("cowardly refusing to save to a terminal. Use the -o flag or redirect")
+		}
+	}
+
+	return SaveImages(ctx, client, images, options.Stdout, options)
```

Comment:
sorry I mean we can remove the `writer` arg in `SaveImages` right? Since `options` already has the `writer` from `options.Stdout`.

I think we should avoid having overlap between args within `options` and other args in a func (e.g., here `writer` v.s. `opt.Stdout`).

Reply:
Yep. I'll remove it.